destination-filters-kotlin/src/main/java/com/segment/analytics/plugins/DestinationFilters.kt

```diff
@@ -54,26 +66,34 @@ class DestinationFilters : Plugin {
                 evaluate(tsubScript)
                 evaluate(tsubEvaluate)
             }
+            engineRecreated = true
         }
 
-        removeExistingFilters()
+        var cacheInvalidated = false
+        if (cachedRules != settings.middlewareSettings["routingRules"]) {
+            cachedRules = settings.middlewareSettings["routingRules"]
+            cacheInvalidated = true
+        }
+
+        // if engine recreated or rules are different we need to remove and recreated filters
+        // otherwise, we can skip
+        if (!engineRecreated && !cacheInvalidated) return
```

Comment:
This comment is confusing because it's talking in the affirmative but the variables are negated. Maybe something like:

// If the engine has NOT been recreated and the cache/rules have not been invalidated then we can exit early.